misc/specs/atrium-specs-jvm/src/main/kotlin/ch/tutteli/atrium/specs/integration/OptionalAssertionsSpec.kt

```diff
@@ -42,4 +43,24 @@ abstract class OptionalAssertionsSpec (
             }
         }
     }
+
+    describeFun(isPresent.name) {
+        val isPresentFun = isPresent.lambda
+
+        context("is present") {
+            it("does not throw") {
+                val isPresentValue: Optional<Any> = Optional.of(2)
+                expect(isPresentValue).isPresentFun(2)
+            }
+
+            it("throws an AssertionError") {
+                expect {
```

Comment:
I suggest you comment this line and line 60-62 and let the spec run. This way you can see how the reporting looks like.
I am sure you will see why the current implementation is not yet as it should be.